src/track/taglib/trackmetadata_common.h

```diff
@@ -0,0 +1,202 @@
+        const QByteArray& byteArray) {
+    if (byteArray.isNull()) {
+        return TagLib::ByteVector::null;
+    } else {
+        return TagLib::ByteVector(byteArray.constData(), byteArray.size());
+    }
+}
+
+#if defined(__EXTRA_METADATA__)
+inline TagLib::String uuidToTString(
+        const QUuid& uuid) {
+    return toTString(uuidToNullableStringWithoutBraces(uuid));
+}
+#endif // __EXTRA_METADATA__
+
+inline QString formatBpm(
+        const TrackMetadata& trackMetadata) {
+    return Bpm::valueToString(
+            trackMetadata.getTrackInfo().getBpm().getValue());
+}
+
+bool parseBpm(
+        TrackMetadata* pTrackMetadata,
+        const QString& sBpm);
+
+inline QString formatReplayGainGain(
+        const ReplayGain& replayGain) {
+    return ReplayGain::ratioToString(replayGain.getRatio());
+}
+
+inline QString formatReplayGainPeak(
+        const ReplayGain& replayGain) {
+    return ReplayGain::peakToString(replayGain.getPeak());
+}
+
+inline QString formatTrackGain(
+        const TrackMetadata& trackMetadata) {
+    return formatReplayGainGain(
+            trackMetadata.getTrackInfo().getReplayGain());
+}
+
+bool parseTrackGain(
+        TrackMetadata* pTrackMetadata,
+        const QString& dbGain);
+
+inline QString formatTrackPeak(
+        const TrackMetadata& trackMetadata) {
+    return formatReplayGainPeak(
+            trackMetadata.getTrackInfo().getReplayGain());
+}
+
+bool parseTrackPeak(
+        TrackMetadata* pTrackMetadata,
+        const QString& strPeak);
+
+#if defined(__EXTRA_METADATA__)
+inline QString formatAlbumGain(
+        const TrackMetadata& trackMetadata) {
+    return formatReplayGainGain(trackMetadata.getAlbumInfo().getReplayGain());
+}
+
+bool parseAlbumGain(
+        TrackMetadata* pTrackMetadata,
+        const QString& dbGain);
+
+inline QString formatAlbumPeak(
+        const TrackMetadata& trackMetadata) {
+    return formatReplayGainPeak(trackMetadata.getAlbumInfo().getReplayGain());
+}
+
+bool parseAlbumPeak(
+        TrackMetadata* pTrackMetadata,
+        const QString& strPeak);
+#endif // __EXTRA_METADATA__
+
+bool parseSeratoMarkers(
+        TrackMetadata* pTrackMetadata,
+        const QByteArray& data);
+
+bool parseSeratoMarkers2(
+        TrackMetadata* pTrackMetadata,
+        const QByteArray& data);
+
+inline QImage loadImageFromByteVector(
+        const TagLib::ByteVector& imageData,
+        const char* format = nullptr) {
+    return QImage::fromData(
+            // char -> uchar
+            reinterpret_cast<const uchar*>(imageData.data()),
+            imageData.size(),
+            format);
+}
+
+// Bitmask of optional tag fields that should NOT be read from the
+// common part of the tag through TagLib::Tag.
+// Usage: The write functions for ID3v2, MP4, APE and XiphComment tags
+// have specialized code for some or all of the corresponding tag fields
+// and the common implementation sometime doesn't work as expected.
+enum ReadTagMask {
```

Comment:
enum class? If they are supposed to be combined we can use `QFlag` for this, like here:
https://github.com/mixxxdj/mixxx/blob/master/src/dialog/dlgreplacecuecolor.h#L24-L31 and https://github.com/mixxxdj/mixxx/blob/master/src/dialog/dlgreplacecuecolor.h#L71